site-manage/datemanager/tool/src/main/java/org/sakaiproject/datemanager/tool/MainController.java

```diff
@@ -47,12 +52,24 @@
 @Controller
 public class MainController {
 
-	@Inject private DateManagerService dateManagerService;
+    @Inject private DateManagerService dateManagerService;
+	
+    @Autowired
+    private SessionManager sessionManager;
+    
+    @Autowired
+    private PreferencesService preferencesService;
+
+    @RequestMapping(value = {"/", "/index"}, method = RequestMethod.GET)
+    public String showIndex(@RequestParam(required=false) String code, Model model, HttpServletRequest request, HttpServletResponse response) {
+
+        String userId = sessionManager.getCurrentSessionUserId();
+        final Locale loc = StringUtils.isNotBlank(userId) ? preferencesService.getLocale(userId) : Locale.getDefault();
+        LocaleResolver localeResolver = RequestContextUtils.getLocaleResolver(request);
+        localeResolver.setLocale(request, response, loc);
 
-	@RequestMapping(value = {"/", "/index"}, method = RequestMethod.GET)
-	public String showIndex(@RequestParam(required=false) String code, Model model) {
-		Locale locale = dateManagerService.getUserLocale();
-		String siteId = dateManagerService.getCurrentSiteId();
+        Locale locale = dateManagerService.getUserLocale();
```

Comment:
Sorry I've been neglecting this PR review 👍 

Why are you bothering to create a new locale object here? You have it already at line 67.

Reply:
Good point, Adrian.
I've made and tested the correction.
Thank you.